### PR TITLE
Fix for False Positive: nested attributes separated by newlines

### PR DESCRIPTION
**Pull request by cbowman0**
Re-use the level logic from the hash section on the array section.

This fixes issue #5

### CODE REVIEW

**Comment by cbowman0** [comment]
I'm not sure how to fix the installation of tin problem on Ruby 1.9.

**Comment by rnelson0** [comment]
I don't know where tins is coming from, but since it's not compliant with ruby <2.0, it's not going to be fixed on the ruby side. If someone knows where it's coming from, perhaps that gem can be removed from the gemspec if it is no longer needed.

**Comment by cbowman0** [comment]
I'm pretty sure it's coveralls.  This fixes it on a server running Ruby 1.9:
```
$ git diff
diff --git a/puppet-lint-trailing_comma-check.gemspec b/puppet-lint-trailing_comma-check.gemspec
index 89a701c..530f12f 100644
--- a/puppet-lint-trailing_comma-check.gemspec
+++ b/puppet-lint-trailing_comma-check.gemspec
@@ -23,6 +23,6 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rspec-collection_matchers', '~> 1.0'
   spec.add_development_dependency 'rspec-json_expectations', '~> 2.0'
   spec.add_development_dependency 'mime-types', '~> 1.0' # 2.x dropped Ruby 1.8 support
-  spec.add_development_dependency 'coveralls', '~> 0.7' unless RUBY_VERSION =~ /^1\.8/
+  spec.add_development_dependency 'coveralls', '~> 0.7' unless RUBY_VERSION =~ /^1\.8|^1\.9/
   spec.add_development_dependency 'rake'
 end
diff --git a/spec/spec_helper.rb b/spec/spec_helper.rb
index b8210a7..3391488 100644
--- a/spec/spec_helper.rb
+++ b/spec/spec_helper.rb
@@ -1,4 +1,4 @@
-unless RUBY_VERSION =~ /^1\.8/
+unless RUBY_VERSION =~ /^1\.8|^1\.9/
   require 'coveralls'
   Coveralls.wear!
 end
```

**Comment by cbowman0** [comment]
Thoughts on this?

**Comment by ffrank** [comment]
Works for me. Ruby 1.9 is 😭 anyway...

**Comment by ekohl** [comment]
Should this be rebased and declared a Ruby 2.0+ gem?

File: spec/puppet-lint/plugins/check_trailing_comma/check_trailing_comma_spec.rb [hunk 1]
```diff
@@ -66,6 +66,19 @@ class { '::apache':
         if $var =~ Sensitive {
           $foo = $var.unwrap
         }
+
+        package { 'foo':
+          require => [
+            File[
+              '/etc/foo.conf',
+              '/etc/bar.conf'
```
Comment (ekohl) [inline spec/puppet-lint/plugins/check_trailing_comma/check_trailing_comma_spec.rb]: I would expect it to require a trailing comma here as well

File: spec/puppet-lint/plugins/check_trailing_comma/check_trailing_comma_spec.rb [hunk 2]
```diff
@@ -66,6 +66,19 @@ class { '::apache':
...
+            ],
+            Packages[
+              'bar',
+              'baz'
```
Comment (ekohl) [inline spec/puppet-lint/plugins/check_trailing_comma/check_trailing_comma_spec.rb]: And here

File: spec/puppet-lint/plugins/check_trailing_comma/check_trailing_comma_spec.rb (hunk 1 above)
Comment (cbowman0) [reply spec/puppet-lint/plugins/check_trailing_comma/check_trailing_comma_spec.rb]: Having a comma at the end of that line is invalid syntax for puppet 3.  Puppet versions 4 and 5 don't care.

Comment (ekohl) [reply spec/puppet-lint/plugins/check_trailing_comma/check_trailing_comma_spec.rb]: Within Voxpupuli we've dropped support for all puppet 3 versions. I'd be happy to do a major version bump with it though.

Comment (cbowman0) [reply spec/puppet-lint/plugins/check_trailing_comma/check_trailing_comma_spec.rb]: Then this PR and bug aren't relevant anymore.  =)

Comment (ekohl) [reply spec/puppet-lint/plugins/check_trailing_comma/check_trailing_comma_spec.rb]: Should we close this then? :)

**Pull request closed without merging**